Showing also financial info from previous years in Info component

tomasfabry opened the pull request:
Implemented https://github.com/vacuumlabs/verejne.digital/issues/171.

borisvida commented on client/src/components/shared/Info/Finances.js:
```diff
@@ -0,0 +1,95 @@
+}
+
+const FinancesItem = ({data, ico}: {data: EnhancedCompanyFinancial, ico: string}) => {  
+  const {employees, profit, profitTrend, revenue, revenueTrend, year} = data
```
You could do `{data: {employees, profit, profitTrend, revenue, revenueTrend, year}, ico}` on L#28 and remove braces and `return` :slightly_smiling_face: Also, I prefer defining `FinancesItemProps` type, it looks less clumsy than defining the type inline.

borisvida replied:
What's your opinion on defining multiple components in one file? I prefer spliting them up, but I know that we do it like this a lot in this project.

tomasfabry replied:
Yes, I absolutely agree with you. I also prefer one component in one file. I've done it this way because I always try to imitate and keep the code style of the project I'm working on ... I will split it.

Just a note. I also prefer to have exactly one scss/css file for each component and for the main style use the same name as component has, i.e. for "Item" it will be ".Item". We used to do it in this way on Dermoi but here it is done in other way... Never mind.

borisvida commented on client/src/components/shared/Info/Finances.js:
```diff
@@ -0,0 +1,95 @@
+}
+
+const Finances = ({data, ico, toggledOn, toggle}: FinancesProps) => {
+  const [lastFinances = {}, ...otherFinances] = data
```
The first note from above is also valid here. If you think it would be unreadable (because of combination of array an object destructuring), I don't insist.

borisvida commented on client/src/components/shared/Info/Finances.js:
```diff
@@ -0,0 +1,95 @@
+            </Button>
+            <div className="older-finances">
+              {toggledOn && otherFinances.map(finances => {
+                return (<FinancesItem data={finances} ico />)
```
Also unnecessary return.

borisvida commented on client/src/components/shared/Info/Finances.js:
```diff
@@ -0,0 +1,95 @@
+export default compose(
+  withState('toggledOn', 'toggle', false),
+  withHandlers({
+    toggle: ({toggle}: StateProps) => () => toggle((current) => !current),
```
The naming here is a bit confusing: `toggledOn` should be maybe calld `open` or `expanded`. There are also two things called `toggle` use in one line, which is really confusing. The first one (defined on L#91) isn't event toggling the variable, it's just setting it. So maybe it should be `setOpen` and `toggleOpen`, respectively. The exact naming is up to you :slightly_smiling_face:

tomasfabry replied:
I've copied this code from `Relations` component as it was which byt the way was not very DRY so I created new component `ToggleBox` where I've implemented your suggestions and I've also used it in the `Relations` component to reuse the code.

borisvida commented on client/src/components/shared/Info/Info.js:
```diff
@@ -53,62 +53,29 @@ type HandlerProps = {
+    finances,
+    ico,
+    terminated_on
+  } = data
```
This again might be desctructured in param and no return would be needed then.

borisvida commented on client/src/components/shared/Info/Finances.js:
```diff
@@ -0,0 +1,95 @@
+  return (
+    <Fragment>
+      <FinancesItem data={lastFinances} ico />
+      {otherFinances.length &&
```
If this is `0`, it will show `0` instead of nothing. Change the check to a boolean, please.

borisvida commented on client/src/components/shared/Info/Finances.js:
```diff
@@ -0,0 +1,95 @@
+            </Badge>
+            </Button>
+            <div className="older-finances">
+              {toggledOn && otherFinances.map(finances => {
```
This is missing a key.

tomasfabry commented:
Thanks for review, I've implemented your recommendations and also some more improvements as I've mentioned above. I've also extracted `Findata` component to stand-alone file.

mirecmrozek commented on client/src/components/shared/ToggleBox.scss:
```diff
@@ -0,0 +1,45 @@
+@import '../../variables.scss';
```
should be just `../variables.scss`
